CHANGELOG.rst

```diff
@@ -7,6 +7,7 @@ Unreleased
 
 * Add closed layers to layer contract.
 * Rename default repository branch to 'main'.
+* Optimise `find_shortest_chains` query.
```

Comment:
Might be worth mentioning that it might lead to slightly different results in some cases? (I agree with you, the results of `find_shortest_chains` is not formally specified, though I'd like us to get to that point.)